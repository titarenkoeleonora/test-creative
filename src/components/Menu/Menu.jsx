import {Link} from "react-router-dom";

const Menu = ({myref}) => {
  return (
    <section className="menu" ref={myref}>
      <div className="menu__wrapper">
        <nav className="menu__navigation">
          <ul className="sidebar menu__sidebar">
            <li className="sidebar__item">
              <Link to="/" className="sidebar__link">
                Главная страница
              </Link>
            </li>
            <li className="sidebar__item">
              <Link to="/movement" className="sidebar__link">
                Функция сдвига элементов вправо на N шагов
              </Link>
            </li>
            <li className="sidebar__item">
              <Link to="/substring" className="sidebar__link">
                Функция нахождения самой длинной последовательности 2х строк
              </Link>
            </li>
            <li className="sidebar__item">
              <Link to="/checkboxes" className="sidebar__link">
                Отображение/скрытие элементов при переключении чек-боксов
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    </section>
  );
};

export default Menu;