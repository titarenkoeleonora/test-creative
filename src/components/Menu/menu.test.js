import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter as Router} from "react-router-dom";

import Menu from "./Menu";

it('Render Menu component', () => {
  const tree = renderer
    .create(
      <Router>
        <Menu />
      </Router>
    );

  expect(tree).toMatchSnapshot();
});