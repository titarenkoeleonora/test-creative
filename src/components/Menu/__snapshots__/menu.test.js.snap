// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Menu component 1`] = `
<section
  className="menu"
>
  <div
    className="menu__wrapper"
  >
    <nav
      className="menu__navigation"
    >
      <ul
        className="sidebar menu__sidebar"
      >
        <li
          className="sidebar__item"
        >
          <a
            className="sidebar__link"
            href="/"
            onClick={[Function]}
          >
            Главная страница
          </a>
        </li>
        <li
          className="sidebar__item"
        >
          <a
            className="sidebar__link"
            href="/movement"
            onClick={[Function]}
          >
            Функция сдвига элементов вправо на N шагов
          </a>
        </li>
        <li
          className="sidebar__item"
        >
          <a
            className="sidebar__link"
            href="/substring"
            onClick={[Function]}
          >
            Функция нахождения самой длинной последовательности 2х строк
          </a>
        </li>
        <li
          className="sidebar__item"
        >
          <a
            className="sidebar__link"
            href="/checkboxes"
            onClick={[Function]}
          >
            Отображение/скрытие элементов при переключении чек-боксов
          </a>
        </li>
      </ul>
    </nav>
  </div>
</section>
`;
